core/src/main/java/org/commonjava/maven/ext/core/state/NexusStagingMavenPluginRemovalState.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2012 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.ext.core.state;
+
+import org.commonjava.maven.atlas.ident.ref.ProjectRef;
+import org.commonjava.maven.atlas.ident.ref.SimpleVersionlessArtifactRef;
+import org.commonjava.maven.ext.annotation.ConfigValue;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Captures configuration relating to nexus-staging-maven-plugin removal from the POMs.
+ */
+public final class NexusStagingMavenPluginRemovalState
+        extends PluginRemovalState
```

Comment:
I can understand reusing the implementation but I'm not clear what benefits you are bringing by extending the PluginRemovalState?

Reply:
Because I needed to shared the common method `getPluginRemoval()` which is called by the `PluginRemovalManipulator` to tell it which plugins to remove.

Reply:
Ah I see. Yes ok that makes sense